interp/run.go

```diff
@@ -124,31 +124,35 @@ func typeAssertStatus(n *node) {
 	c0, c1 := n.child[0], n.child[1]
 	value := genValue(c0)              // input value
 	value1 := genValue(n.anc.child[1]) // returned status
-	typ := c1.typ.rtype                // type to assert
+	typ := c1.typ                      // type to assert
 	next := getExec(n.tnext)
 
 	switch {
-	case c0.typ.cat == valueT:
+	case isInterfaceSrc(c1.typ):
```

Comment:
```suggestion
	case isInterfaceSrc(typ):
```

Reply:
Instead, to make clear that we test the type to assert, and not the input type, we could keep using `c1.type` everywhere except in the closure (spare an indirection for performance). A comment will be added to specify that `c0` contains the input value and `c1.typ` the type to assert.